src/start.sh

```diff
@@ -103,31 +108,31 @@ start() {
     echo "  [i] FTL log output is disabled. Remove the Environment variable TAIL_FTL_LOG, or set it to 1 to enable FTL log output."
   fi
 
-  # Wait for the ftl process to finish and handle its return
-  wait $ftl_pid
-  stop $?
+  # Wait for the capsh process (which spawned FTL) to finish
+  wait $CAPSH_PID
+  
+  # If we are here, then FTL has exited.
+  # If the trap was triggered, then stop will have already been called
+  if [ $TRAP_TRIGGERED -eq 0 ]; then
+    # Pass the exit code through to the stop function
+    stop $?
+  fi
 }
 
 stop() {
   local FTL_EXIT_CODE=$1
 
-  # Only attempt to close pihole-FTL if it is running, it may already have crashed
-  if pgrep pihole-FTL >/dev/null; then
+  # if we have nothing in FTL_EXIT_CODE, then have been called by the trap. Close FTL and wait for the CAPSH_PID to finish
+  if [ -z "${FTL_EXIT_CODE}" ]; then
+    TRAP_TRIGGERED=1
     echo ""
     echo "  [i] Container stop requested..."
     echo "  [i] pihole-FTL is running - Attempting to shut it down cleanly"
     echo ""
-    # Ensure pihole-FTL shuts down cleanly on SIGTERM/SIGINT
-    ftl_pid=$(pgrep pihole-FTL)
-
     killall --signal 15 pihole-FTL
```

Comment:
Wondering if it's worth doing something like:

```bash
(sleep 3 && killall --signal 15 pihole-FTL) &
wait $CAPSH_PID
```

in case the $CAPSH_PID is already dead by the time we get to it after killing FTL? Or even

```bash
killall --signal 15 pihole-FTL &
wait $CAPSH_PID
```

Am I just being overly paranoid about slower systems here?